src/julia.h

```diff
@@ -179,6 +179,20 @@ STATIC_INLINE int jl_array_ndimwords(uint32_t ndims)
     return (ndims < 3 ? 0 : ndims-2);
 }
 
+typedef struct {
+    union {
```

Comment:
The type is marked as not being pointer-free, despite being immutable,
which seems to work fine. In any case, I don't think this is likely to be
what's causing the problem here.

On Monday, May 11, 2015, Jameson Nash <notifications@github.com> wrote:

> In src/julia.h
> <https://github.com/JuliaLang/julia/pull/11235#discussion_r30087499>:
>
> > @@ -179,6 +179,20 @@ STATIC_INLINE int jl_array_ndimwords(uint32_t ndims)
> >      return (ndims < 3 ? 0 : ndims-2);
> >  }
> >
> > +typedef struct {
> > +    union {
>
> the GC won't see the pointer (or the bytevec), in general, if it is
> declared immutable. putting it inside another object (such as an
> ASCIIString) is sufficient to hide it from GC tracking.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/JuliaLang/julia/pull/11235/files#r30087499>.
>